packages/plugin-ext/src/plugin/comments.ts

```diff
@@ -172,11 +177,30 @@ export class CommentsExtImpl implements CommentsExt {
 
         const documentData = this._documents.getDocumentData(URI.revive(uriComponents));
         if (documentData) {
-            const ranges: theia.Range[] | undefined | null = await commentController.commentingRangeProvider!.provideCommentingRanges(documentData.document, token);
-            if (ranges) {
-                return ranges.map(x => fromRange(x));
+            const commentingRanges:
+                | theia.Range[]
+                | theia.CommentingRanges
+                | undefined
+                | null =
+                await commentController.commentingRangeProvider!.provideCommentingRanges(
+                    documentData.document,
+                    token
+                );
+            if (isArray(commentingRanges)) {
+                return {
+                    ranges: commentingRanges.map(x => fromRange(x)),
+                    fileComments: false
+                };
+            } else if (commentingRanges) {
+                return {
+                    ranges: commentingRanges.ranges?.map(x => fromRange(x)) || [],
+                    fileComments: commentingRanges.enableFileComments
+                };
+            } else {
+                return commentingRanges ?? undefined;
```

Comment:
We already now that `!commentingRanges` here, why not just return `undefined`?

Reply:
indeed, being null or undefined, we don't care as we want undefined only. Removed in next commit.